### PR TITLE
feat(onesync): add CVehicleScriptGameStateDataNode research

### PR DESCRIPTION
**Pull request by PichotM**

### CODE REVIEW

File: code/components/citizen-server-impl/include/state/SyncTrees_Five.h [hunk 1]
```diff
@@ -430,7 +430,213 @@ struct CVehicleGameStateDataNode
...
+		int v24 = 0;
+
+		do
+		{
+			bool hasVehicleWeapon = state.buffer.ReadBit();
+			if (hasVehicleWeapon)
+			{
+				int capacity = state.buffer.Read<int>(11);
+			}
+		} while (v24 < 6);
```
Comment (blattersturm) [inline code/components/citizen-server-impl/include/state/SyncTrees_Five.h:561]: nice loop

**Comment by MRV6** [comment]
#1233

File: code/components/citizen-server-impl/include/state/SyncTrees_Five.h (hunk 1 above)
Comment (PichotM) [reply code/components/citizen-server-impl/include/state/SyncTrees_Five.h:561]: oops lol

**Comment by PichotM** [comment]
> #1233

lol missed that

**Pull request closed without merging**